app/models/user_standing.rb

```diff
@@ -0,0 +1,12 @@
+class UserStanding < ApplicationRecord
+  belongs_to :user
+  belongs_to :standing
+  belongs_to :creator, class_name: "User"
+  belongs_to :archiver, class_name: "User", optional: true
+
+  validates :reason, presence: true
+  validates_with RateLimitValidator,
+                 limit: 100,
+                 scope: :creator_id,
+                 time_frame: 1.hour
```

Comment:
Keep a larger rate limit. We will be using API from the app engine and it is highly possible to report more than 100 standings in an hour.

Reply:
Shouldn't this be scoped to the user as well? That way we can keep a sensible limit? Like 10 per hour or something?

Reply:
> We will be using API from the app engine and it is highly possible to report more than 100 standings in an hour.

Case: 1

I think we may process more than 100 submissions. Because it takes at least 1 minute to run the GitHub action and at least some minutes completing plagiarism check.

The time frame is 60 minutes.

 Case 2:
 
 It will cross if we directly create user standing without any checks.
 
 > Keep a larger rate limit.
 
 Will keep it 1000
 
 ------
 
 > Shouldn't this be scoped to the user as well? That way we can keep a sensible limit? Like 10 per hour or something?
 
 Actually, I thought about this but ignored. Will have a look again.

Reply:
> I think we may not process more than 100 submissions. Because it takes at least 1 minute to run the GitHub action and at least some minutes completing plagiarism check.

100 students can easily submit at a time and GA runs the actions parallel as each of them would be independent repos. 


> Shouldn't this be scoped to the user as well? That way we can keep a sensible limit? Like 10 per hour or something?

I think we should have a hard limit on the number of standing a user can have;

Reply:
> I think we should have a hard limit on the number of standing a user can have

That also works. Say 100 max log entries? 🤷🏼‍♂️